internal/pkg/bulk/engine.go

```diff
@@ -115,36 +115,46 @@ func NewBulker(es esapi.Transport, tracer *apm.Tracer, opts ...BulkOpt) *Bulker
 	}
 
 	return &Bulker{
-		opts:                  bopts,
-		es:                    es,
-		ch:                    make(chan *bulkT, bopts.blockQueueSz),
-		blkPool:               sync.Pool{New: poolFunc},
-		apikeyLimit:           semaphore.NewWeighted(int64(bopts.apikeyMaxParallel)),
-		tracer:                tracer,
-		remoteOutputConfigMap: make(map[string]map[string]interface{}),
-		// remote ES bulkers
-		bulkerMap: make(map[string]Bulk),
+		opts:        bopts,
+		es:          es,
+		ch:          make(chan *bulkT, bopts.blockQueueSz),
+		blkPool:     sync.Pool{New: poolFunc},
+		apikeyLimit: semaphore.NewWeighted(int64(bopts.apikeyMaxParallel)),
+		tracer:      tracer,
 	}
 }
 
 func (b *Bulker) GetBulker(outputName string) Bulk {
-	return b.bulkerMap[outputName]
+	o, ok := b.bulkerMap.Load(outputName)
+	if !ok {
+		return nil
+	}
+	return o.(Bulk)
 }
 
+// GetBulkerMap returns a copy of the BulkerMap
 func (b *Bulker) GetBulkerMap() map[string]Bulk {
-	return b.bulkerMap
```

Comment:
From what I can see, we had a flaky test because the policy self-monitor (`internal/pkg/policy/self.go`) was getting the bulker map in order to check remote output health; but we did not prevent our policy output preparation from creating a new bulker concurrently (`internal/pkg/policy/policy_output.go`).
I've changes our maps to `sync.Map` as we weren't properly using the mutex we had, and changed this func to return a copy instead